libs/resiliency/include/hpx/resiliency/async_replay.hpp

```diff
@@ -183,45 +185,20 @@ namespace hpx { namespace resiliency {
     }
 
     ///////////////////////////////////////////////////////////////////////////
-    /// Asynchronously launch given function \a f. Repeat launching
-    /// on error exactly \a n times (except if abort_replay_exception is thrown).
+    // Asynchronously launch given function \a f. Repeat launching
+    // on error exactly \a n times (except if abort_replay_exception is thrown).
     template <typename F, typename... Ts>
     hpx::future<
         typename hpx::util::detail::invoke_deferred_result<F, Ts...>::type>
-    async_replay(std::size_t n, F&& f, Ts&&... ts)
+    tag_invoke(async_replay_t, std::size_t n, F&& f, Ts&&... ts)
     {
-        return async_replay_validate(n, detail::replay_validator{},
-            std::forward<F>(f), std::forward<Ts>(ts)...);
-    }
-
-    ///////////////////////////////////////////////////////////////////////////
-    /// Functional version of \a hpx::resiliency::async_replay
-    namespace functional {
+        using result_type =
+            typename hpx::util::detail::invoke_deferred_result<F, Ts...>::type;
 
-        struct async_replay_validate
```

Comment:
Out of curiousity, why are we getting rid of the functional variants? I think they can still be useful.

Reply:
The CPOs _are_ functions objects, so we use those instead (e.g. see here: https://github.com/STEllAR-GROUP/hpx/pull/4851/files#diff-db2a88f97b940ef2953ded38a9b07291R36)

Reply:
Got it.